core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -273,8 +276,19 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
 
     override def processNotification(notificationMessage: String) {
       val resource: Resource = Resource.fromString(notificationMessage)
-      val acls = getAclsFromZk(resource)
-      updateCache(resource, acls)
+      try {
+        val acls = getAclsFromZk(resource)
+        updateCache(resource, acls)
+      } catch {
+        case e: InterruptedException =>
+          // It is OK to have InterruptedException after authorizer is closed.
+          if (!isClosed.get)
+            throw e
+          authorizerLogger.debug(s"NotificationHandler is interrupted after authorizer is closed")
+          Set.empty
+        case e: Exception =>
+          throw e
```

Comment:
Good point. I thought about this but wasn't sure if scala handles it in this way. Will fix it.

Thanks!